collector/zfs_linux.go

```diff
@@ -27,31 +27,41 @@ import (
 )
 
 const (
-	zfsArcstatsProcpath = "spl/kstat/zfs/arcstats"
+	zfsProcpathBase = "spl/kstat/zfs/"
 )
 
-func (c *zfsCollector) openArcstatsFile() (file *os.File, err error) {
-	file, err = os.Open(procFilePath(zfsArcstatsProcpath))
+var pathMap = map[zfsSubsystemName]string{
```

Comment:
This can be made part of the zfsCollector struct, right? Then we avoid further polluting the global namespace.